### PR TITLE
Completed docker and kubernetes config

### PR DESCRIPTION
**Pull request by tomfitzherbert**
- Docker now runs as non-root
- Added keycloak
- Integrated Lets Encrypt certs
- Drone updated to also include prod deployment
- Added session table to db startup checks

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,12 +3,14 @@ LABEL maintainer="thomas.fitzherbert1@homeoffice.gsi.gov.uk"
 
 ENV DOCKER_HOME /usr/src/app
 
-RUN mkdir -p ${DOCKER_HOME}
+
+
```
Comment (chrisns) [inline Dockerfile]: lots of whitespace?

File: README.md [hunk 2]
```diff
@@ -46,11 +46,8 @@ The application is designed to be deployed in Kubernetes and expects the
 PostgreSQL database to be provisioned separately to the `kubectl` scripts
 (usually in RDS).
 
-```
-kubectl create -f ./kube/service.yml
-kubectl create -f ./kube/deployment.yml
-kubectl create -f ./kube/ingress.yml
-```
+The `kubenetes` scripts included in this repo are only intended for use in the
```
Comment (chrisns) [inline README.md]: ~~kubenetes~~ kubernetes

Comment (chrisns) [reply README.md]: also, not true, they're intended to be executed with [kd](https://github.com/ukhomeoffice/kd)

File: Dockerfile (hunk 1 above)
Comment (tomfitzherbert) [reply Dockerfile]: Whitespace removed

File: README.md (hunk 2 above)
Comment (tomfitzherbert) [reply README.md]: Comments updated to reflect changes to use `kd`

**Pull request merged**